src/lib/stores/filter_store.ts

```diff
@@ -0,0 +1,28 @@
+import { writable } from 'svelte/store';
+
+import type { Filters } from 'src/types/filters.type';
+
+import { FilterRepository } from '../repositories/filter_repository';
+import { filterListener } from './filtered_items_store';
+
+const currentFilter: Filters = {
+	easy: false,
+	search: ''
+};
+
+export const filterStore = writable<Filters>(currentFilter);
+
+const filterRepository = new FilterRepository(filterStore);
+
+export const easyHandler = () => {
+	filterRepository.updateEasyFilter(currentFilter);
+	console.log(currentFilter);
+	filterListener(currentFilter);
```

Comment:
Is there any better way to do this? I feel like there's probably some kind of functionality to listen to changes without the need of these "etc Listener()"

Reply:
Yes. You can create a `derived` store, which comes from Svelte's `derived` function. 

Their docs example is [this](https://svelte.dev/docs#run-time-svelte-store-derived): 
```ts
import { derived } from 'svelte/store';
// a is an existing store
const doubled = derived(a, $a => $a * 2);
```

I'll post in the `filtered_items_store.ts` comments how you can update that.